lib/reek/smells/prima_donna_method.rb

```diff
@@ -28,18 +28,25 @@ def self.contexts # :nodoc:
 
       def examine_context(ctx)
         ctx.node_instance_methods.map do |method_sexp|
-          next unless method_sexp.ends_with_bang?
+          check_for_smells(method_sexp, ctx)
+        end.compact
+      end
 
-          version_without_bang = ctx.node_instance_methods.find do |sexp_item|
-            sexp_item.name.to_s == method_sexp.name_without_bang
-          end
-          next if version_without_bang
+      private
 
-          SmellWarning.new self,
-                           context: ctx.full_name,
-                           lines: [ctx.exp.line],
-                           message: "has prima donna method `#{method_sexp.name}`"
-        end.compact
+      def check_for_smells(method_sexp, ctx)
+        return unless method_sexp.ends_with_bang?
```

Comment:
I’m on the fence whether this check should be a part of the method (in general, I’d say yes) or – in this particular case – remove it from this method and move the check to the caller:

```ruby
check_for_smells(method_sexp, ctx) unless method_sexp.ends_with_bang?
```

Thoughts?